common/src/main/java/com/microsoft/identity/common/internal/controllers/CommandDispatcher.java

```diff
@@ -132,6 +135,17 @@ private static void cleanMap(BaseCommand command) {
         sExecutingCommandMap = newMap;
     }
 
+    public static void configureCommandDispatcher(CommandDispatcherConfiguration config){
```

Comment:
Since this method is public, we're probably safest to synchronize it.

Reply:
Done.  Thanks!

Reply:
OK, now that we've put the writes to this reference under the class-level monitor, we have to do the same thing to all the reads.

Reply:
Sorry Adam.  Would you mind being a bit more verbose in your feedback.  Is the suggestion that we need to synchronize access to the static variable for the thread pool?  Why would need to be synchronized on reads in this case?  Also, what are you suggesting that we need to prevent.  This is not a public method that intended to be called by anyone else than us.  So not part of the public API... 

Strictly speaking this should be set once per process and never changed.  I wasn't prepared at this point to turn command dispatcher into singleton, but I think that's the more correct approach.

Reply:
What I'm saying is that, yes, you should synchronize all access to the static reference, and under the same monitor.  I'd like to avoid any self-enforcing singletons going forward, but that's a bigger configuration->client construction task than I think we're prepared to tackle here, so I'm OK with it.  But this really means finding the place (I think it's one, maybe two) where the access to the static variable is taken and make sure that it's under the same monitor (in this case, the class) where it is written.

And this isn't about necessarily protecting us from our clients, though that helps, it's protecting us from our future selves.  I mean, spotBugs will flag this anyway as inconsistent synchronization.